### PR TITLE
Feature: Helm kubernetes Deployment

### PR DESCRIPTION
**Pull request by alexohneander**
Added helm deployment so you can quickly deploy it to your Kubernetes cluster.

The deployment consists of:
 - Service
 - Deployment
 - PVC
 - Ingress
 - ServiceAccount

### CODE REVIEW

File: charts/gitness/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          volumeMounts:
+            - name: datadir
+              mountPath: /data
+          livenessProbe:
+            httpGet:
+              path: /
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /
```
Comment (abhinav-harness) [inline charts/gitness/templates/deployment.yaml:49]: We can add `api/v1/system/version` as endpoint for health and readiness check

File: charts/gitness/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        {{- include "gitness.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "gitness.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
```
Comment (abhinav-harness) [inline charts/gitness/templates/deployment.yaml:30]: We will probably need dind-daemon to run for us to support running CI too. I tested it like this 

```        - name: dind-daemon
          image: docker:24.0.3-dind
          command: ["dockerd", "--host", "tcp://127.0.0.1:2375"]
          resources:
            requests:
              cpu: 20m
              memory: 512Mi
          lifecycle:
            postStart:
              exec:
                command:
                  - "/bin/sh"
                  - "-c"
                  - "echo '127.0.0.1 host.docker.internal' >> /etc/hosts"
          securityContext:
            privileged: true
          volumeMounts:
            - name: docker-data
              mountPath: /certs
```

also added 
```
          env:
            - name: DOCKER_HOST
              value: tcp://localhost:2375
```

in gitness container

Comment (alexohneander) [reply charts/gitness/templates/deployment.yaml:30]: @abhinav-harness Do you want to add this as 2 containers or as a second pod that can then be accessed from gitness?

File: charts/gitness/templates/deployment.yaml (hunk 1 above)
Comment (alexohneander) [reply charts/gitness/templates/deployment.yaml]: I have added these probes. Of course, it is much better if we request an endpoint.

**Comment by arya-harness** [comment]
It would be good to add a NOTES.txt file for post install/upgrade steps. eg. Where to access the endpoint.

File: charts/gitness/templates/deployment.yaml (hunk 2 above)
Comment (alexohneander) [reply charts/gitness/templates/deployment.yaml:30]: @abhinav-harness I added it. The container can be deactivated via the Values files and the image can be exchanged via it.

**Comment by alexohneander** [comment]
@arya-harness I have added. This is actually created but I had to add with `git add -f`.

**Comment by willzhang** [comment]
thanks for your work, can you add env to helm charts, i want define password when install
```
  -e GITNESS_PRINCIPAL_ADMIN_PASSWORD=gitness \
  -e GITNESS_USER_SIGNUP_ENABLED=false \
```

and the values.yaml maybe like this
```
env: {}

extraEnvVars: []
```

**Comment by alexohneander** [comment]
Why is this PR Closed?

**Pull request closed without merging**